pkg/controlplane/controller/legacytokentracking/controller.go

```diff
@@ -0,0 +1,191 @@
+		AddFunc: func(obj interface{}) {
+			configMap := obj.(*corev1.ConfigMap)
+			c.enqueue(configMap)
+		},
+		UpdateFunc: func(oldObj, newObj interface{}) {
+			configMap := newObj.(*corev1.ConfigMap)
+			c.enqueue(configMap)
+		},
+		DeleteFunc: func(obj interface{}) {
+		},
+	})
+
+	return c
+}
+
+func (c *Controller) enqueue(configMap *corev1.ConfigMap) {
+	key, err := cache.MetaNamespaceKeyFunc(configMap)
+	if err != nil {
+		klog.Errorf("Failed to get key for ConfigMap: %v", configMap)
+		return
+	}
+	c.queue.Add(key)
+}
+
+// Run starts the controller sync loop.
+func (c *Controller) Run(stopCh <-chan struct{}) {
+	defer utilruntime.HandleCrash()
+	defer c.queue.ShutDown()
+
+	klog.Info("Starting legacy_token_tracking_controller")
+	defer klog.Infof("Shutting down legacy_token_tracking_controller")
+
+	go c.configMapInformer.Run(stopCh)
+	if !cache.WaitForNamedCacheSync("configmaps", stopCh, c.configMapSynced) {
+		return
+	}
+
+	go wait.Until(c.runWorker, time.Second, stopCh)
+
+	c.queue.Add(metav1.NamespaceSystem + "/" + ConfigMapName)
+
+	<-stopCh
+	klog.Info("Ending legacy_token_tracking_controller")
+}
+
+func (c *Controller) runWorker() {
+	for c.processNext() {
+	}
+}
+
+func (c *Controller) processNext() bool {
+	key, quit := c.queue.Get()
+	if quit {
+		return false
+	}
+	defer c.queue.Done(key)
+
+	if err := c.syncConfigMap(key.(string)); err != nil {
+		utilruntime.HandleError(fmt.Errorf("while syncing ConfigMap %q, err: %w", key, err))
+		c.queue.AddRateLimited(key)
+		return true
+	}
+	c.queue.Forget(key)
+	return true
+}
+
+func (c *Controller) syncConfigMap(key string) error {
+	obj, exists, err := c.configMapCache.GetByKey(key)
+	if err != nil {
+		return err
+	}
+
+	switch {
+	case c.enabled:
+		if !exists {
+			if !c.creationRatelimiter.Allow() {
+				c.queue.AddAfter(key, c.creationRatelimiter.Reserve().Delay())
+				return nil
+			}
+
+			if _, err = c.configMapClient.ConfigMaps(metav1.NamespaceSystem).Create(context.TODO(), &corev1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{Namespace: metav1.NamespaceSystem, Name: ConfigMapName},
+				Data:       map[string]string{ConfigMapDataKey: time.Now().Format(time.RFC3339)},
+			}, metav1.CreateOptions{}); !apierrors.IsAlreadyExists(err) {
+				return err
+			}
+		} else {
+			configMap := obj.(*corev1.ConfigMap)
+			if _, err = time.Parse(time.RFC3339, configMap.Data[ConfigMapDataKey]); err != nil {
+				configMap.Data[ConfigMapDataKey] = time.Now().Format(time.RFC3339)
+				if _, err = c.configMapClient.ConfigMaps(metav1.NamespaceSystem).Update(context.TODO(), configMap, metav1.UpdateOptions{}); err != nil {
+					return err
+				}
+			}
+		}
+
+	case !c.enabled:
+		if exists {
+			if err := c.configMapClient.ConfigMaps(metav1.NamespaceSystem).Delete(context.TODO(), ConfigMapName, metav1.DeleteOptions{}); err != nil {
+				return err
+			}
+		}
+	}
+	c.queue.AddAfter(key, 5*time.Minute) // polling every five minute to make sure the configmap is in desired state.
```

Comment:
prefer reacting to delete events rather than polling here

Reply:
removed